tests/route/test_default_route.py

```diff
@@ -44,18 +57,22 @@ def test_default_route_set_src(duthost):
         pytest.fail("cannot find ipv6 Loopback0 address")
 
     rt = duthost.command("ip route list match 0.0.0.0")['stdout_lines']
-    m = re.match(r"^default proto bgp src (\S+)", rt[0])
+    m = re.match(r"^default proto (bgp|186) src (\S+)", rt[0])
     if m:
-        if ipaddress.ip_address(m.group(1)) != lo_ipv4.ip:
+        if ipaddress.ip_address(m.group(2)) != lo_ipv4.ip:
             pytest.fail("default route set src to wrong IP {} != {}".format(m.group(1), lo_ipv4.ip))
     else:
         pytest.fail("default route do not have set src. {}".format(rt))
 
     rt = duthost.command("ip -6 route list match ::")['stdout_lines']
     m = re.match(r"^default proto bgp src (\S+)", rt[0])
+    m1 = re.match(r"default via (\S+) dev (\S+) proto 186 src (\S+)", rt[0])
     if m:
         if ipaddress.ip_address(m.group(1)) != lo_ipv6.ip:
             pytest.fail("default route set src to wrong IP {} != {}".format(m.group(1), lo_ipv6.ip))
+    elif m1:
+        if ipaddress.ip_address(m1.group(3)) != lo_ipv6.ip:
+            pytest.fail("default route set src to wrong IP {} != {}".format(m1.group(3), lo_ipv6.ip))
```

Comment:
These 2 lines can be merged into a single line:

assertions.pytest_assert(ipaddress.ip_address(m1.group(3)) == lo_ipv6.ip, "default route set src to wrong IP {} != {}".format(m1.group(3), lo_ipv6.ip))

There are more opportunities to condense code above and below in this file.

You do need to import, but this is one line for all the assertions you can use in this file.
from common.helpers import assertions

Reply:
I feel current way is more readable

Reply:
actually, it cannot because one of m and m1 is none. they cannot be use as the same time.

Reply:
you could
assertions.pytest_assert(not m or condition 1, msg1)
asertsions.pytest_assert(not m1 or condition 2, msg2) 

or you can keep you current
if m:
    assertion 1
if m1:
    assertion 2

Reply:
prefer to keep it is

Reply:
changed.